core/src/kura.rs

```diff
@@ -79,15 +75,19 @@ impl Kura {
             broker,
             block_reciever: Mutex::new(block_reciever),
             block_sender,
-            block_thread_handle: Mutex::new(None),
         });
 
+        Ok(kura)
+    }
+
+    /// Start kura thread
+    pub fn start(kura: &Arc<Self>) -> ThreadHandler {
         // How do we avoid the spawned thread taking on a
         // life of it's own, aka keeping kura alive and running
         // long after it has been dropped by all who needed it.
         // By using a weak reference, see `kura_recieve_blocks_thread`
         // for details on how this works.
-        let kura_weak_reference = Arc::downgrade(&kura);
+        let kura_weak_reference = Arc::downgrade(kura);
```

Comment:
The weak reference was the original shutdown mechanism. Since you are replacing it we can revert to just using an Arc. This simplifies the code inside the kura thread loop since we don't need to upgrade the reference.